indy_node/server/request_handlers/config_req_handlers/auth_rule/abstract_auth_rule_handler.py

```diff
@@ -29,8 +33,24 @@ def _static_validation_for_rule(self, operation, identifier, req_id):
         StaticAuthRuleHelper.check_auth_key(operation, identifier, req_id, self.write_req_validator.auth_map)
 
     def _update_auth_constraint(self, auth_key: str, constraint):
+        version = self.database_manager.state_version
+        if version in self._update_state_by_versions:
+            self._update_state_by_versions[version](auth_key, constraint)
+        else:
+            self.state.set(AbstractAuthRuleHandler.make_state_path_for_auth_rule(auth_key),
+                           self.constraint_serializer.serialize(constraint))
+
+    def _update_auth_constraint_for_1_9_1(self, auth_key: str, constraint):
+        constraint_dict = constraint.as_dict if constraint else {}
+        self._set_off_ledger_to_constraint(constraint_dict)
         self.state.set(AbstractAuthRuleHandler.make_state_path_for_auth_rule(auth_key),
-                       self.constraint_serializer.serialize(constraint))
+                       self.constraint_serializer.serializer.serialize(constraint_dict))
+
+    def _set_off_ledger_to_constraint(self, constraint_dict):
+        if constraint_dict[CONSTRAINT_ID] == ConstraintsEnum.ROLE_CONSTRAINT_ID:
+            constraint_dict[OFF_LEDGER_SIGNATURE] = None
```

Comment:
Why do we set it to None here? I think it needs to be just explicitly set regardless of if this is True or False (now it's explicitly set if it's True only).